src/pip/_internal/req/constructors.py

```diff
@@ -325,7 +325,8 @@ def install_req_from_req_string(
         PyPI.file_storage_domain,
         TestPyPI.file_storage_domain,
     ]
-    if req.url and comes_from.link.netloc in domains_not_allowed:
+    if req.url and comes_from and comes_from.link and \
```

Comment:
My preference would be to use implied line continuation with parentheses rather than backslash (as in PEP 8 as well).

Reply:
Done